dali/pipeline/operators/fused/crop_cast_permute.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef DALI_PIPELINE_OPERATORS_FUSED_CROP_CAST_PERMUTE_H_
+#define DALI_PIPELINE_OPERATORS_FUSED_CROP_CAST_PERMUTE_H_
+
+#include <vector>
+#include <utility>
+
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+
+namespace dali {
+
+template <typename Backend>
+class CropCastPermute : public Operator<Backend> {
+ public:
+  explicit inline CropCastPermute(const OpSpec &spec) :
+    Operator<Backend>(spec),
+    output_type_(spec.GetArgument<DALIDataType>("output_dtype")),
+    output_layout_(spec.GetArgument<DALITensorLayout>("output_layout")),
+    image_type_(spec.GetArgument<DALIImageType>("image_type")),
+    color_(IsColor(image_type_)),
+    C_(color_ ? 3 : 1) {
+    vector<int> temp_crop;
+    GetSingleOrDoubleArg(spec, &temp_crop, "crop");
```

Comment:
Please merge with the newest master - this function is no longer there (use `GetSingleOrrepeatedArg` instead).